test/OpenTelemetry.Tests/Implementation/Trace/Export/BatchingActivityProcessorTests.cs

```diff
@@ -274,21 +277,36 @@ public async Task ShutdownOnNotEmptyQueueFullFlush()
                                         .SetSampler(new AlwaysOnActivitySampler())
                                         .AddProcessorPipeline(pp => pp.AddProcessor(ap => activityProcessor)));
 
-            var activities = new List<Activity>();
-            for (int i = 0; i < 100; i++)
+            using (var testListener = new TestEventListener())
             {
-                activities.Add(this.CreateActivity(i.ToString()));
-            }
+                // Enable events from OpenTelemetrySDK.
+                testListener.EnableEvents(OpenTelemetrySdkEventSource.Log, EventLevel.Verbose);
+                var activities = new List<Activity>();
+                for (int i = 0; i < 100; i++)
+                {
+                    activities.Add(this.CreateActivity(i.ToString()));
+                }
 
-            Assert.True(activityExporter.ExportedActivities.Length < activities.Count);
-            using (var cts = new CancellationTokenSource(DefaultTimeout))
-            {
-                await activityProcessor.ShutdownAsync(cts.Token);
-            }
+                Assert.True(activityExporter.ExportedActivities.Length < activities.Count);
+                using (var cts = new CancellationTokenSource(DefaultTimeout))
+                {
+                    await activityProcessor.ShutdownAsync(cts.Token);
+                }
 
-            Assert.True(activityExporter.WasShutDown);
-            Assert.Equal(activities.Count, activityExporter.ExportedActivities.Length);
-            Assert.InRange(exportCalledCount, activities.Count / batchSize, activities.Count);
+                // Get the shutdown event.
+                var shutdownEvent = testListener.Events.Where((e) => e.EventId == 2).First();
```

Comment:
"2" is not magic. Its the eventid for the shutdownevent used by the BatchingProcessor.

Reply:
Might need to make this an internal enum and make it visible to test project.